### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Filienko**
Updated updated docker to python 3.11 (however, the libraries are generated by pip-compile available for python 3.9)

Manually pinned fhirclient to 4.1.0 to prevent date errors that occur in the newest FHIRDate, which has a different date requirements from the one we currently utilize in backend isacc.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,43 +1,133 @@
...
+    # via flask
+cryptography==43.0.0
+    # via
+    #   authlib
+    #   python-jose
+ecdsa==0.19.0
+    # via python-jose
+exceptiongroup==1.2.2
+    # via cattrs
 fhirclient==4.1.0
```
Comment (ivan-c) [inline requirements.txt:50]: I saw in 07ea20f you had to downgrade from `4.2.0`. 

Could you add a pin (and comment) in `setup.cfg` so it doesn't get overwritten next time `pip-compile` is run?

Please also note (in a comment, PR description, anywhere) any other "manual" changes you had to make to requirements.* after running `pip-compile`

**Review by ivan-c — APPROVED** [review]
Looks good, thank you!

File: isacc_messaging/models/isacc_fhirdate.py [hunk 2]
```diff
@@ -35,5 +35,5 @@ def __repr__(self):
         return self.isostring
 
 
-DEEP_PAST = IsaccFHIRDate("1975-01-01T00:00:00Z")
-DEEP_FUTURE = IsaccFHIRDate("2025-01-01T00:00:00Z")
```
Comment (ivan-c) [inline isacc_messaging/models/isacc_fhirdate.py]: Many datetime libraries use the current ("local") timezone if one is not specified, which can make[ tests non-deterministic](https://martinfowler.com/articles/nonDeterminism.html) (ie success depends the timezone)

Is it possible to change the datetime format, so a timezone is always explicitly included?

**Review by ivan-c — APPROVED** [review]
Looks great! We'll want to test on isacc dev beforehand, but everything looks good so far

File: isacc_messaging/models/isacc_fhirdate.py (hunk 2 above)
Comment (Filienko) [reply isacc_messaging/models/isacc_fhirdate.py]: Accidental delete, reverted to original format!

**Pull request merged**